### DISCUSSION TITLE
10. October 2023

### DISCUSSION THREAD

**Discussion by dimakuv** [gramineproject/gramine#1597] [Meeting notes]
### Agenda

*(please write your proposed agenda items in comments under this discussion)*

- [x] woju: packaging and releasing MAA attestation plugin from contrib repo (gramineproject/contrib#57)
- [x] Mona and Woju: Gramine website content review
- [x] Dmitrii & Mona: "Secret Provisioning" name for a library is pretty bad; it's frequently used even without provisioning the secrets, just to run before the main application. So maybe we should re-brand it, and maybe even split into several libraries (e.g. one "active" pre-main lib, one "passive" secret-prov interfaces one).

---

Greeting Stefan Berger, who submitted several reviews and PRs recently!

### Woju: packaging and releasing MAA attestation plugin from contrib repo

PR: https://github.com/gramineproject/contrib/pull/57

Woju's proposal is to add a sub-repository `contrib` to our Gramine repo. If user adds this sub-repo, then the MAA plugin package appears and can be installed.

Woju explained that `contrib` is a standard term in Debian repos. The other ones are `non-free`, `universe`, `multiverse` -- but MAA doesn't fit there.

Woju: wants to do it in v1.6. No objections from people present on the call (but @mkow was sick, so need his ack).

Kailun: we don't have any documentation pointing to this package. How will people learn that this exists?
- **TODO(Dmitrii)**: Prepare a PR based on the issue https://github.com/gramineproject/gramine/issues/1569.

### Mona & Woju: Gramine website content review

Mona showed the old web site and the draft for a new web site. The new web site will look like https://projectacrn.org/.

### Dmitrii & Mona: "Secret Provisioning" rename/split

Dmitrii described 3 different usages of the SecretProv library:
1. Pre-main constructor-time secret provisioning of a single secret; no modifications to the app.
2. Higher-level API than RA-TLS (think of EasyCurl instead of OpenSSL), where one func connects to the server and gets the secret; minimal modifications to the app: 1-2 function calls.
3. "Pro" version of (2), like Curl vs EasyCurl; more modifications to the app: 4-5 function calls (or whatever the app wants to do).

Suggestion: rename to "Trust Manager" (?).

Suggestion: split into two libs: 1 ("Premain") and 2-3 ("Live").

**Comment by monavij** [comment]
Gramine website content review - we have a draft  - Mona and Woju
